homeassistant/components/onkyo/config_flow.py

```diff
@@ -0,0 +1,235 @@
+"""Config flow for Onkyo."""
+from __future__ import annotations
+
+import asyncio
+from typing import Any
+
+import async_timeout
+from pyeiscp import Connection
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+
+from .const import (
+    CONF_ENABLED_SOURCES,
+    CONF_IDENTIFIER,
+    CONF_MAX_VOLUME,
+    CONF_SOURCES,
+    CONNECT_TIMEOUT,
+    DEFAULT_MAX_VOLUME,
+    DEFAULT_SOURCE_NAMES,
+    DEFAULT_SOURCES,
+    DISCOVER_TIMEOUT,
+)
+from .const import DOMAIN  # pylint:disable=unused-import
+
+
+class OnkyoConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
```

Comment:
Import of the existing YAML configuration is missing?
We should migrate existing users of the integration.

Reply:
I'm working on this now, am I correct that I should implement the `async_step_import` in the config flow? And should this directly create entries with the options from the old config?

Reply:
Correct :)

Reply:
Next question, how do I create multiple entries from one flow? All examples I see return self.async_create_entry, which only creates a single entry.

Reply:
> Next question, how do I create multiple entries from one flow?

You don't. Instead, call start the import flow multiple times.

Reply:
How do I start the import flow, isn't this done automatically when there is an entry in the yaml file?
In the current onkyo integration, all configuration variables are optional, so this is a valid yaml entry:
```
media_player:
  - platform: onkyo
```
which will discover and add all available receivers. How do I start multiple import flows from this one line?

Reply:
Here is an example on how to import an entry.

https://github.com/home-assistant/core/blob/606989917954b6549efa35f56a589f07357c57e8/homeassistant/components/stookalert/binary_sensor.py#L36-L57

`hass.config_entries.flow.async_init` can be called multiple times if needed.

Reply:
Thanks a lot for the help! So I should add this specific part to media_player.py?